libs/libcommon/include/common/arithmeticOverflow.h

```diff
@@ -113,9 +113,7 @@ inline bool mulOverflow(__int128 x, __int128 y, __int128 & res)
     if (!x || !y)
         return false;
 
-    unsigned __int128 a = (x > 0) ? x : -x;
-    unsigned __int128 b = (y > 0) ? y : -y;
-    return (a * b) / b != a;
+    return res/x != y; /// whether overflow int128
```

Comment:
How about add some unit test for this overflow check?

Reply:
added